model/hex.py

```diff
@@ -47,21 +47,25 @@ def range(self, *args):
         for diff in hexes_range(*args):
             yield self + diff
 
-    def on_line(self, other, obstacles):
+    def on_line(self, other, speed, path):
         '''
-        Returns if this hex is one one line with the other and if there i obstacle between them.
+        Returns if this hex is one one line with the other and if there is no obstacle between them.
         
         <param name="other">Other hex.</param>
         '''
 
         if self.q == other.q or self.r == other.r or self.s == other.s:
-            # Yes enemy is in range but let's check if there is obstacle between them
-            if is_obstacle_betweenX(self, other, obstacles) or is_obstacle_betweenY(self, other, obstacles) or is_obstacle_betweenZ(self, other, obstacles):
+            # Other is aligned to the hex, let's check if there is obstacle between them
+            if self.is_obstacle_between(other, speed, path):
```

Comment:
I believe obstacle checking should be on game level, not in hex method

Reply:
Maybe. Will change.